spatialmath/box.go

```diff
@@ -352,3 +356,82 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector.
+func (b *box) ToPointCloud(resolution float64) []r3.Vector {
+	b.pose.Orientation()
+
+	// check for user defined spacing
+	var iter float64
+	if resolution != 0. {
+		iter = resolution
+	} else {
+		iter = defaultBoxPointDensity
+	}
+	var faces [][]float64
+	// create points on box faces with box centered at (0, 0, 0)
+
+	fillFaces := func(fixedDimension int) {
+		starts := [3]float64{0.0, 0.0, 0.0}
+		starts[fixedDimension] = b.halfSize[fixedDimension]
+		for i := starts[0]; i <= b.halfSize[0]; i += iter {
+			for j := starts[1]; i <= b.halfSize[1]; i += iter {
+				for k := starts[2]; k <= b.halfSize[2]; k += iter {
+					p1 := []float64{i, j, k}
+					p2 := []float64{i, j, -k}
+					p3 := []float64{i, -j, k}
+					p4 := []float64{i, -j, -k}
+					p5 := []float64{-i, j, k}
+					p6 := []float64{-i, j, -k}
+					p7 := []float64{-i, -j, -k}
+					p8 := []float64{-i, -j, k}
+
+					switch {
+					case i == 0.0 && j == 0.0:
+						faces = append(faces, p1, p2)
+					case j == 0.0 && k == 0.0:
+						faces = append(faces, p1, p5)
+					case i == 0.0 && k == 0.0:
+						faces = append(faces, p1, p7)
+					case i == 0.0:
+						faces = append(faces, p1, p2, p3, p4)
+					case j == 0.0 && fixedDimension == 0:
+						faces = append(faces, p1, p2, p5, p6)
+					case k == 0.0:
+						faces = append(faces, p1, p3, p5, p8)
+					case i != 0.0 && j != 0.0 && k != 0.0:
+						faces = append(faces, p1, p2, p3, p4, p5, p6, p7, p8)
+					}
+				}
+			}
+		}
+	}
+
+	// TODO: the for loop below can be made concurrent if the ToPointCloud method is too slow
+	for i := 0; i < 3; i++ {
+		fillFaces(i)
+	}
+	myList := transformPointsToPose(faces, b.Pose())
+	return myList
+}
+
+// transformPointsToPose multiples each point by the rotation matrix then adds the displacement.
+func transformPointsToPose(points [][]float64, pose Pose) []r3.Vector {
+	translateBy := []float64{pose.Point().X, pose.Point().Y, pose.Point().Z}
```

Comment:
converted back to former function using the `Compose` method to handle transformations

Reply:
Why was this swapped back to no longer use Compose?